db/crud.go

```diff
@@ -1312,76 +1312,93 @@ func (db *DatabaseCollectionWithUser) PutExistingCurrentVersion(ctx context.Cont
 					if err != nil {
 						return nil, nil, false, nil, err
 					}
-				} else {
-					// temp remove history alignment here, pending CBG-4791
-					alignRevTrees = false
+					// the new document has a dominating hlv, so we can ignore any legacy rev revtree information on the incoming document
 					revTreeConflictChecked = true
-					previousRevTreeID = doc.GetRevTreeID()
-					newGeneration = prevGeneration + 1
-					base.InfofCtx(ctx, base.KeyCRUD, "conflict detected between the two HLV's for doc %s, incoming version %v, local version %v", base.UD(doc.ID), newDocHLV.ExtractCurrentVersionFromHLV(), doc.HLV.ExtractCurrentVersionFromHLV())
-					if conflictResolver.hlvConflictResolver == nil {
-						// cancel rest of update, HLV is in conflict and no resolver is present
-						return nil, nil, false, nil, base.HTTPErrorf(http.StatusConflict, "Document revision conflict")
+					if !alignRevTrees {
+						previousRevTreeID = doc.GetRevTreeID()
+					} else {
+						// align rev tree here for ISGR replications
+						err = doc.alignRevTreeHistoryForHLVWrite(ctx, db, newDoc, revTreeHistory)
+						if err != nil {
+							return nil, nil, false, nil, err
+						}
 					}
-					// resolve conflict
-					var newHLV *HybridLogicalVector
-					newHLV, err = db.resolveHLVConflict(ctx, doc, newDoc, conflictResolver.hlvConflictResolver)
-					if err != nil {
-						base.InfofCtx(ctx, base.KeyCRUD, "Failed to resolve HLV conflict for doc %s, error: %v", base.UD(doc.ID), err)
-						return nil, nil, false, nil, err
+				case HLVConflict:
+					// if the legacy rev list is from ISGR property, we should not do a conflict check
+					if len(revTreeHistory) > 0 && !alignRevTrees {
+						// TODO: CBG-4828 - conflict resolution for legacy rev documents
+						// conflict check on rev tree history, if there is a rev in rev tree history we have the parent of locally we are not in conflict
+						parent, currentRevIndex, err = db.revTreeConflictCheck(ctx, revTreeHistory, doc, newDoc.Deleted)
+						if err != nil {
+							base.InfofCtx(ctx, base.KeyCRUD, "conflict detected between the two HLV's for doc %s, incoming version %s, local version %s, and conflict found in rev tree history", base.UD(doc.ID), newDocHLV.GetCurrentVersionString(), doc.HLV.GetCurrentVersionString())
+							return nil, nil, false, nil, err
+						}
+						_, err = doc.addNewerRevisionsToRevTreeHistory(newDoc, currentRevIndex, parent, revTreeHistory)
+						if err != nil {
+							return nil, nil, false, nil, err
+						}
+						revTreeConflictChecked = true
+						doc.HLV.UpdateWithIncomingHLV(newDocHLV)
+						if err != nil {
+							return nil, nil, false, nil, err
+						}
+					} else {
+						revTreeConflictChecked = true
+						base.InfofCtx(ctx, base.KeyCRUD, "conflict detected between the two HLV's for doc %s, incoming version %v, local version %v", base.UD(doc.ID), newDocHLV.ExtractCurrentVersionFromHLV(), doc.HLV.ExtractCurrentVersionFromHLV())
```

Comment:
```suggestion
						base.DebugfCtx(ctx, base.KeyCRUD, "conflict detected between the two HLV's for doc %s, incoming version %v, local version %v", base.UD(doc.ID), newDocHLV.ExtractCurrentVersionFromHLV(), doc.HLV.ExtractCurrentVersionFromHLV())
```

At least drop to debug, maybe even KeyVV for testing. Did we log conflict behavior for revtrees?

Reply:
Dropped to debug